dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -108,12 +110,24 @@ public BooleanFormulaManager getBooleanFormulaManager() {
         return booleanFormulaManager;
     }
 
+    public BooleanFormula encodeBooleanExpression(Event event, ExprInterface expression) {
+        return new ExpressionEncoder(formulaManager, event).encodeBoolean(expression);
+    }
+
+    public Formula encodeIntegerExpression(Event event, ExprInterface expression) {
+        return new ExpressionEncoder(formulaManager, event).encodeInteger(expression);
+    }
+
+    public BooleanFormula encodeAtom(COpBin op, Formula lhs, Formula rhs) {
```

Comment:
`Atom` is just another `BExpr`, why not simply make `encodeBooleanExpression` cover this case? Also, why do the other two take an event as input but not this one? Where is the difference?

Reply:
This method is provided for `AssertBasic`, which requires no event.  Also, it accepts both operands in their encoded form.

Reply:
Maybe you should rename the methods with a suffix "At": `encodeBooleanExpressionAt`.
If you then reverse the order of parameters, I think it fits nicely.

You also might want to rename `encodeAtom`. The issue is that we use `Atom` to refer to an expression class, yet this function does not take an `Atom` as input.
Maybe rename it to `encodeAtomicFormula`, `encodeFormulaAtom`, or even `encodeFormulaComparison`.